PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,52 +1,75 @@
-## Which issue is fixed by this PR? If not related to an existing issue, what does this PR do?
+## What does this PR (Pull Request) do?
+<!-- Explain the changes you made here. If this fixes an Issue, identify it by
+number e.g.
 
-Fixes #(replace_this_text_with_issue_number) 
+Fixes #9001
```

Comment:
Is the example better than the replace? I worry it won't be replaced properly.

Reply:
I see a lot of PRs just leave that `Fixes #(replace_this_text_with_issue_number)` and add stuff beneath it. This way we won't have that clutter when a PR doesn't relate to an issue.

Reply:
I think people do often forget to link the PR to an issue. In my template PR, I changed it from 'Fixes #(issue_number)' to  'Fixes #(replace_this_text_with_issue_number)' hoping to get more people to fill in the blank. Maybe we should also make this a checklist option? Just throwing an idea out there:

This PR fixes an issue #(Issue Number Here)
This PR adds a new feature
Other

Reply:
If we make those three checkboxes, we'll be up to a total of 9 and a maximum of 7 will ever be checked. I don't think that's a viable solution, because we'd be back to incomplete checklists.

TBH, I don't think `Fixes #(issue_number)` was really unclear, I suspect the vast majority of those who didn't actually replace that text made that choice out of sheer laziness/forgot to delete it after describing what the PR did and it wasn't tied to an issue.

Reply:
Some issues aren't "fixes" per se, they could just be "enhancements".
How about "This PR is for issue #(issue number here) OR does not match an existing issue"?

Reply:
I do believe that would break the default functionality of closing issues when a PR mentions that it "Fixes"/"Resolves"/"Closes" an Issue. It's true that not all issue-related PRs will necessarily close said issue, and maybe I should make not of that, but I'm hesitant to disable that nice little automatic housekeeping - I think it'll probably result in a ton of unclosed issues that have actually been resolved.

Reply:
Then "This PR resolves issue #(issue number here) OR does not match an existing issue (insert GH issue URL here)"

Reply:
Is that `(insert GH issue URL here)` meant to be replaced by a new Issue created by the PR author? If so, maybe the checklist line should just look like:

```markdown
[ ] This PR resolves issue #(INSERT ISSUE NUMBER HERE) <!-- If an issue does not already exist, please create one -->
```

Reply:
No I meant `https://github.com/apache/trafficcontrol/issues` but was too lazy to get that URL. This is what I think it should be:
```
[ ] This PR resolves issue #(INSERT ISSUE NUMBER HERE) or does not match an existing issue (see https://github.com/apache/trafficcontrol/issues)
```

Reply:
Well, before I make that change, I just want to be sure you know that the example in that snippet is in a comment, so the default PR doesn't include an issue link with replacement text at all. The context isn't large enough for you to see that, unfortunately.

Reply:
I don't think we should require that all PRs have a matching issue. However, contributors should still be encouraged to do a quick search for any matching existing issues so that we can close them out if the PR resolves the issue. By checking the box you're pretty much saying you've searched for an existing issue.

Reply:
I like the checkbox idea

`[ ] This PR resolves issue #(INSERT ISSUE NUMBER HERE) or does not match an existing issue (see https://github.com/apache/trafficcontrol/issues)`

Reply:
To be clear, I don't think a PR should require an issue either. But I agree that asking authors to search for issues is a good idea.